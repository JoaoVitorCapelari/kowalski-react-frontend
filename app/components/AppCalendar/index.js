/**
*
* AppCalendar based on bulma-calendar
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import format from 'date-fns/format';
import startOfMonth from 'date-fns/start_of_month';
import startOfWeek from 'date-fns/start_of_week';
import isSameDay from 'date-fns/is_same_day';
import isSameMonth from 'date-fns/is_same_month';
import isWithinRange from 'date-fns/is_within_range';
import addDays from 'date-fns/add_days';
import getDate from 'date-fns/get_date';
import filter from 'lodash/filter';
import FaChevronLeft from 'react-icons/lib/fa/chevron-left';
import FaChevronRight from 'react-icons/lib/fa/chevron-right';

const extractDaysForCalendar = (date) => {
  const extractedStartOfMonth = startOfMonth(date);
  const weeks = [];

  let currentStartOfWeek = startOfWeek(extractedStartOfMonth);
  do {
    for (let i = 0; i < 7; i += 1) {
      weeks.push(addDays(currentStartOfWeek, i));
    }
    currentStartOfWeek = addDays(currentStartOfWeek, 7);
  } while (isSameMonth(currentStartOfWeek, date));

  return weeks;
};

/*
 * highlightedRanges: An array of pairs, where the first is the start of the range and the second it's end.
 * tooltips: An object mapping from dates to tooltip messages
 * disabledDates: An array containing dates that should appear as disabled despite the fact the they are within the current reference month
 */
const renderCalendarDay = (
  referenceDate, { tooltips = {}, highlightedRanges = [], disabledDates = [] }, dateClickCallback, day
) => {
  const isWithinCurrentMonth = isSameMonth(day, referenceDate);
  const isActive = isSameDay(day, referenceDate);
  const tooltip = filter(tooltips, (k) => isSameDay(k, day))[0];
  const isStartOfRange = highlightedRanges.some((range) => isSameDay(day, range[0]));
  const isEndOfRange = highlightedRanges.some((range) => isSameDay(day, range[1]));
  const withinRange = isStartOfRange || isEndOfRange || highlightedRanges.some((range) => isWithinRange(day, range[0], range[1]));

  const outerClassNameSufixes = [
    isWithinCurrentMonth ? '' : ' is-disabled',
    tooltip ? ' tooltip' : '',
    withinRange ? ' calendar-range' : '',
    isStartOfRange ? ' range-start' : '',
    isEndOfRange ? ' range-end' : '',
  ];

  const innerClassNameSufixes = [
    isActive ? ' is-active' : '',
    isStartOfRange || isEndOfRange ? ' is-active' : '',
  ];

  const outerElementProperties = {
    className: `calendar-date ${outerClassNameSufixes.join('')}`,
    key: day.getTime(),
  };

  const innerElementProperties = {
    className: `date-item ${innerClassNameSufixes.join('')}`,
  };

  if (tooltip) {
    outerElementProperties['data-tooltip'] = tooltip;
  }

  if (dateClickCallback) {
    innerElementProperties.onClick = () => dateClickCallback(day);
  }

  if (disabledDates.some((date) => isSameDay(day, date))) {
    outerElementProperties.disabled = '';
  }

  return (
    <div {...outerElementProperties}>
      <button {...innerElementProperties}>
        {getDate(day)}
      </button>
    </div>
  );
};

const renderCalendarDays = (date, options, dateClickCallback) => extractDaysForCalendar(date)
  .map(renderCalendarDay.bind(this, date, options, dateClickCallback));
const extractMonthAndYearExpression = (day) => format(day, 'MMMM GGGG');

function AppCalendar({ selectedDate, options = {}, onNextMonthClicked, onPreviousMonthClicked, onDateClicked }) {
  return (
    <div className="calendar">
      <div className="calendar-nav">
        <div className="calendar-nav-left">
          <button className="button is-text" onClick={onPreviousMonthClicked}>
            <FaChevronLeft />
          </button>
        </div>
        <div className="yearMonth" >{extractMonthAndYearExpression(selectedDate)}</div>
        <div className="calendar-nav-right">
          <button className="button is-text" onClick={onNextMonthClicked}>
            <FaChevronRight />
          </button>
        </div>
      </div>
      <div className="calendar-container">
        <div className="calendar-header">
          <div className="calendar-date">Sun</div>
          <div className="calendar-date">Mon</div>
          <div className="calendar-date">Tue</div>
          <div className="calendar-date">Wed</div>
          <div className="calendar-date">Thu</div>
          <div className="calendar-date">Fri</div>
          <div className="calendar-date">Sat</div>
        </div>
        <div className="calendar-body">
          { renderCalendarDays(selectedDate, options, onDateClicked) }
        </div>
      </div>
    </div>
  );
}

AppCalendar.propTypes = {
  selectedDate: PropTypes.instanceOf(Date).isRequired,
  options: PropTypes.object,
  onNextMonthClicked: PropTypes.func,
  onPreviousMonthClicked: PropTypes.func,
  onDateClicked: PropTypes.func,
};

export default AppCalendar;
