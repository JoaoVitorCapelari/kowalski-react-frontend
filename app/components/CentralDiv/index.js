/**
*
* CentralDiv
*
*/

import styled from 'styled-components';

const CentralDiv = styled.div`
  display: flex;
  flex-grow: 1;
  align-items: center;
  justify-content: center;
`;

export default CentralDiv;
