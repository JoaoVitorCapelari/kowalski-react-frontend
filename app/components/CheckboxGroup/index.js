/**
*
* CheckboxGroup
*
*/

import * as React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { Field } from 'redux-form/immutable';

const CheckboxContainer = styled.div`
  display: flex;
  flex-wrap: wrap;
  border-radius: 4px;
  border: 1px solid #aaa;
  padding-left: 10px;
  margin-top: 10px;
`;

export default class CheckboxGroup extends React.Component {
  item = ({ input, label }) => (
    <label htmlFor={input.id}>
      <input type="checkbox" {...input} />
      <span>{label}</span>
    </label>
  );

  handleChange = ({ input, input: { onChange, onBlur }, value }) => (event) => {
    const inputValue = input.value;
    const arr = [...inputValue];
    if (event.target.checked) {
      arr.push(value);
    } else {
      arr.splice(arr.indexOf(value), 1);
    }
    onBlur(arr);
    return onChange(arr);
  };

  field = ({ input, meta, options }) => {
    const { name, onFocus } = input;
    const { component = this.item } = this.props;
    const inputValue = input.value;

    const checkboxes = options.map((option, index) => {
      const { label, value } = option;
      const key = `checkbox-${index}`;

      return React.createElement(component, {
        input: {
          name: `${name}[${index}]`,
          checked: inputValue.includes(value),
          onChange: this.handleChange({ input, value }),
          value,
          onFocus,
        },
        meta,
        label,
        option,
        options,
        key,
      });
    });

    return <CheckboxContainer>{checkboxes}</CheckboxContainer>;
  };

  render() {
    return <Field {...this.props} component={this.field} />;
  }
}

CheckboxGroup.propTypes = {
  /* eslint-disable react/no-unused-prop-types */
  options: PropTypes.arrayOf(PropTypes.shape({
  /* eslint-enable react/no-unused-prop-types */
    label: PropTypes.node.isRequired,
    value: PropTypes.node.isRequired,
  })).isRequired,
  component: PropTypes.func,
};
