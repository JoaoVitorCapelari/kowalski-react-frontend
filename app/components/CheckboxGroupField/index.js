/**
*
* CheckboxGroup
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const CheckboxContainer = styled.div`
  display: flex;
  flex-wrap: wrap;
  border-radius: 4px;
  border: 1px solid #aaa;
  padding-left: 10px;
  margin-top: 10px;
`;

// TODO export this component to another file
const DefaultCheckboxGroupItem = ({ input, label }) => (
  <label htmlFor={input.id}>
    <input type="checkbox" {...input} />
    <span>{label}</span>
  </label>
);

DefaultCheckboxGroupItem.propTypes = {
  input: PropTypes.shape({
    id: PropTypes.any.isRequired,
  }).isRequired,
  label: PropTypes.string.isRequired,
};

const createChangeHandler = ({ input, input: { onChange, onBlur }, value }) => (event) => {
  // TODO change the code to handle value as a List from immutable
  const arr = [...input.value];
  if (event.target.checked) {
    arr.push(value);
  } else {
    arr.splice(arr.indexOf(value), 1);
  }
  onBlur(arr);
  return onChange(arr);
};

const CheckboxGroupField = ({ input, meta, options, checkboxItemComponent }) => {
  const { name, onFocus } = input;
  const inputValue = input.value;
  const Component = checkboxItemComponent || DefaultCheckboxGroupItem;
  const checkboxes = options.map((option, index) => {
    const { label, value } = option;
    const key = `checkbox-${index}`;

    return (
      <Component
        input={{
          name: `${name}[${index}]`,
          id: `${name}[${index}]`,
          checked: inputValue.includes(value),
          onChange: createChangeHandler({ input, value }),
          value,
          onFocus,
        }}
        {...{
          meta,
          label,
          option,
          options,
          key,
        }}
      />
    );
  });

  return <CheckboxContainer>{checkboxes}</CheckboxContainer>;
};

CheckboxGroupField.propTypes = {
  options: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.node.isRequired,
    value: PropTypes.node.isRequired,
  })).isRequired,
  input: PropTypes.shape({
    name: PropTypes.string,
    onFocus: PropTypes.func,
  }),
  meta: PropTypes.object.isRequired,
  checkboxItemComponent: PropTypes.func,
};

export default CheckboxGroupField;
