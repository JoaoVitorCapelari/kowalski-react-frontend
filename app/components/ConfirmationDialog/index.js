/**
*
* ConfirmationDialog
*
*/

import React from 'react';
import PropTypes from 'prop-types';

function ConfirmationDialog({ message, onCancel, onConfirm, isActive }) {
  return (
    <div className={`modal ${isActive ? 'is-active' : ''}`}>
      <div className="modal-background"></div>
      <div className="modal-card">
        <header className="modal-card-head">
          <p className="modal-card-title">Confirmation</p>
          <button className="delete" aria-label="close"></button>
        </header>
        <section className="modal-card-body">
          { message }
        </section>
        <footer className="modal-card-foot">
          <button className="button is-success" onClick={onConfirm}>Confirm</button>
          <button className="button" onClick={onCancel}>Cancel</button>
        </footer>
      </div>
    </div>
  );
}

ConfirmationDialog.propTypes = {
  message: PropTypes.string.isRequired,
  onCancel: PropTypes.func.isRequired,
  onConfirm: PropTypes.func.isRequired,
  isActive: PropTypes.bool.isRequired,
};

export default ConfirmationDialog;
