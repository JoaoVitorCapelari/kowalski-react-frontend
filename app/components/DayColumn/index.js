/**
*
* DayColumn
*
*/

import React from 'react';
import styled from 'styled-components';
import format from 'date-fns/format';
import PropTypes from 'prop-types';
import { List } from 'immutable';
import parseAmountOfHours from '../../support/parsers/duration';

const wrapperFlexBasis = 14.286;
const slotsBackgroundColor = '#fff';

const Wrapper = styled.div`
  flex-grow: 0;
  flex-basis: ${wrapperFlexBasis}%;
  display: flex;
  flex-direction: column;
  align-items: stretch;
`;

const DayLabelWrapper = styled.div`
  text-align: center;
`;

const SlotsContainer = styled.div`
  flex-basis: ${100 - wrapperFlexBasis}%;
  background-color: ${slotsBackgroundColor};
  border: 0.5px solid #aaa;
  border-right:0px;
  display: flex;
  flex-direction: column;
`;

const generateSlotWrapper = (hourSlot, totalSlot) => styled.a`
  display: flex;
  flex-direction: column;
  flex-basis: ${(hourSlot * 100) / totalSlot}%;
  justify-content: center;
  text-align: center;
  padding: 10px;
  flex-grow: 0;
`;

const generatePlusSlotWrapper = (hourSlot, totalSlot) => generateSlotWrapper(hourSlot, totalSlot).extend`
  display: inline-flex;
  flex-direction: column;
  justify-content: center;
  text-align: center;
  font-size: 0;
  flex-grow: 1;

  &:hover {
    font-size: 3rem;
  }
`;

const renderTimeSlots = (timeSlots) => {
  return (timeSlots || List()).map((timeSlot, index) => {
    const amountOfHours = parseAmountOfHours(timeSlot.get('duration'));
    const SlotWrapper = generateSlotWrapper(amountOfHours, 8.0);
    // TODO put the id of the log instead of index. This is going to be returned by the server,
    // so we must wait until it's integrated
    return (
      <SlotWrapper key={index}>
        <p>{ timeSlot.get('duration') }</p>
        <p>{ timeSlot.get('project') } - { timeSlot.get('activity') }</p>
      </SlotWrapper>
    );
  })
};

const amountOfHoursInTimeSlots = (timeSlots) => {
  return (timeSlots || List()).reduce((memo, current) => {
    return memo + parseAmountOfHours(current.get('duration'));
  }, 0.0);
}

const renderPlusButton = (timeSlots, onClickCallback) => {
  const amountOfHours = 8.0 - amountOfHoursInTimeSlots(timeSlots);
  const SlotWrapper = generatePlusSlotWrapper(amountOfHours, 8.0);

  return (
    <SlotWrapper role="button" onClick={onClickCallback} tabIndex={0}>+</SlotWrapper>
  );
}

function DayColumn({ day, onFreeSlotClick, timeSlots }) {
  return (
    <Wrapper>
      <DayLabelWrapper>
        <p>{ format(day, 'ddd') }</p>
        <p>{ format(day, 'D') }</p>
      </DayLabelWrapper>
      <SlotsContainer>
        { renderTimeSlots(timeSlots) }
        { renderPlusButton(timeSlots, onFreeSlotClick) }
      </SlotsContainer>
    </Wrapper>
  );
}

DayColumn.propTypes = {
  day: PropTypes.instanceOf(Date).isRequired,
  onFreeSlotClick: PropTypes.func,
  timeSlots: PropTypes.instanceOf(List),
};

export default DayColumn;
