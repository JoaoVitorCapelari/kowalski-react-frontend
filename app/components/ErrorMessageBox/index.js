/**
*
* ErrorMessageBox
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Container = styled.div`
  display: flex;
  flex-grow: 1;
`;

function ErrorMessageBox({ errorMsg }) {
  if (!errorMsg) {
    return <div></div>;
  }

  return (
    <Container className="message is-danger">
      <Container className="message-body">
        { errorMsg }
      </Container>
    </Container>
  );
}

ErrorMessageBox.propTypes = {
  errorMsg: PropTypes.string,
};

export default ErrorMessageBox;
