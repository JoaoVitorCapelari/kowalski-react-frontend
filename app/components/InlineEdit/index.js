/**
*
* InlineEdit
*
*/

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import FaPencil from 'react-icons/lib/fa/pencil';
import FaCheck from 'react-icons/lib/fa/check';
import TiTimes from 'react-icons/lib/ti/times';
import { ClipLoader } from 'react-spinners';

const Container = styled.div`
  display: flex;
  flex-grow: 1;

  ${({ stateName }) => (
    [STATE_HOVERED, STATE_EDITING].includes(stateName) ? `
      outline: 1px solid grey;
    ` : ''
  )}
`;

const ContentContainer = styled.div`
  display: flex;
  flex-grow: 1;
  position: relative;
  width: 100%;
`;

const ButtonsContainer = styled.a`
  display: none;
  color: inherit;
  margin-right: 5px;
  flex-grow: 0;

  ${({ stateName }) => (
    [STATE_HOVERED, STATE_SAVING].includes(stateName) ? `
      display: block;
    ` : ''
  )}
`;

const EditActionsWrapper = styled.div`
  display: flex;
  flex-grow: 0;
  position: absolute;
  bottom: -28px;
  right: 0px;
`;

const EditButton = styled.div`
`;

const CheckChangesButton = styled.a`
  display: flex;
  color: inherit;
  margin-left: 2px;
  margin-right: 1px;
`;

const DiscardChangesButton = styled.a`
  display: flex;
  color: inherit;
  margin-left: 1px;
  margin-right: 2px;
`;

export const STATE_NORMAL = 'STATE_NORMAL';
export const STATE_HOVERED = 'STATE_HOVERED';
export const STATE_EDITING = 'STATE_EDITING';
export const STATE_SAVING = 'STATE_SAVING';

const TextBeingSaved = styled.span`
  font-style: italic;
  color: grey;
`;

// TODO Refactor to be a presentational component, should not have internal state
// InlineEdit =>
//  Presentational component
//  Props:
//    - state(saving, normal, hovered, editing)
//    - value
//    - onCommit onDiscard onChange
//    - children: Funcao para renderizacao do componente de edicao ()
// InlineEdit =>
//  Presentational component
//  Props:
class InlineEdit extends Component {
  constructor(props) {
    super(props);
    this.state = { name: props.saving ? STATE_SAVING : STATE_NORMAL, content: props.value };

    this.unhover = this.unhover.bind(this);
    this.hover = this.hover.bind(this);
    this.edit = this.edit.bind(this);
    this.save = this.save.bind(this);
    this.contentChanged = this.contentChanged.bind(this);
    this.dicardChanges = this.dicardChanges.bind(this);
    this.commitChanges = this.commitChanges.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    // Whenever a new value is received, update the content
    if (this.props.value !== nextProps.value) {
      this.setState({
        ...this.state,
        content: nextProps.value,
      });
    }

    if (!!nextProps.saving !== (this.state.name === STATE_SAVING)) {
      this.setState({
        ...this.state,
        name: nextProps.saving ? STATE_SAVING : STATE_NORMAL,
        content: nextProps.saving ? this.state.content : this.props.value,
      });
    }
  }

  contentChanged(content) {
    this.setState({ ...this.state, content });
  }

  unhover() {
    if (this.state.name === STATE_HOVERED) {
      this.setState({ ...this.state, name: STATE_NORMAL });
    }
  }

  dicardChanges() {
    this.setState({ name: STATE_NORMAL, content: this.props.value });
  }

  commitChanges() {
    this.props.onCommit(this.state.content);
  }

  hover() {
    if (this.state.name === STATE_NORMAL) {
      this.setState({ ...this.state, name: STATE_HOVERED });
    }
  }

  edit() { this.setState({ ...this.state, name: STATE_EDITING }); }
  save() { this.setState({ ...this.state, name: STATE_SAVING }); }

  formatValue() {
    return this.props.formatter ? this.props.formatter(this.props.value) : this.props.value;
  }

  renderButtonsContainerForEditingState(stateName) {
    return (
      <EditActionsWrapper stateName={stateName}>
        <CheckChangesButton title="Confirm changes" onMouseDown={this.commitChanges}>
          <FaCheck size={19} />
        </CheckChangesButton>
        <DiscardChangesButton title="Discard changes" onMouseDown={this.dicardChanges}>
          <TiTimes size={22} />
        </DiscardChangesButton>
      </EditActionsWrapper>
    );
  }

  renderEditComponent() {
    return this.props.renderEditComponent({
      content: this.state.content,
      onCommit: this.commitChanges,
      onDiscard: this.dicardChanges,
      onChange: this.contentChanged,
      stateName: this.state.name,
    });
  }

  renderContentOnSaving() {
    if (this.props.displayEditElementWhenSaving) {
      return <TextBeingSaved>{this.renderEditComponent()}</TextBeingSaved>;
    }

    return (
      <TextBeingSaved>
        {
          this.props.formatter
            ? this.props.formatter(this.state.content)
            : this.state.content
        }
      </TextBeingSaved>
    );
  }

  renderButtonsContainerNonEditingState() {
    return (
      <EditButton>
        { this.state.name !== STATE_SAVING && <FaPencil size={18} /> }
        { this.state.name === STATE_SAVING && <ClipLoader size={18} /> }
      </EditButton>
    );
  }

  renderContentOnHovered() {
    const formattedValue = this.formatValue();
    if (this.props.displayEditElementWhenOnHover) {
      return this.renderEditComponent();
    }

    return formattedValue;
  }

  renderContentOnNormalState() {
    const formattedValue = this.formatValue();
    if (this.props.displayEditElementWhenNormal) {
      return this.renderEditComponent();
    }

    return formattedValue;
  }

  render() {
    return (
      <Container
        onMouseEnter={this.hover}
        onMouseLeave={this.unhover}
        stateName={this.state.name}
        onClick={this.edit}
        title={'Click for edit'}
      >
        <ContentContainer>
          { this.state.name === STATE_NORMAL && this.renderContentOnNormalState() }
          { this.state.name === STATE_HOVERED && this.renderContentOnHovered() }
          { this.state.name === STATE_EDITING && this.renderEditComponent() }
          { this.state.name === STATE_EDITING && this.renderButtonsContainerForEditingState(this.state.name) }
          { this.state.name === STATE_SAVING && this.renderContentOnSaving() }
        </ContentContainer>
        <ButtonsContainer stateName={this.state.name}>
          { this.state.name !== STATE_EDITING && this.renderButtonsContainerNonEditingState() }
        </ButtonsContainer>
      </Container>
    );
  }
}

InlineEdit.propTypes = {
  formatter: PropTypes.func, // Optional function to format the value shown
  onCommit: PropTypes.func.isRequired,
  saving: PropTypes.bool,
  renderEditComponent: PropTypes.func.isRequired,
  displayEditElementWhenSaving: PropTypes.bool,
  displayEditElementWhenOnHover: PropTypes.bool,
  displayEditElementWhenNormal: PropTypes.bool,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.instanceOf(Date)]),
};

export default InlineEdit;
