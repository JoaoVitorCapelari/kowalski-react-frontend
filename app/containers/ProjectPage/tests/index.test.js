import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter, Route } from 'react-router-dom';
import { fromJS } from 'immutable';
import { IntlProvider, injectIntl } from 'react-intl';

import { ProjectPage } from '../index';
import { STATE_NORMAL } from '../../../components/InlineEdit';
import { translationMessages } from '../../../i18n';

describe('<ProjectPage />', () => {
  it('renders', () => {
    const ProjectPageWithIntl = injectIntl(ProjectPage);
    const renderedComponent = mount(
      <MemoryRouter initialEntries={['projects/23']}>
        <Route
          exact
          path="projects/:code"
          render={(props) => (
            <IntlProvider locale="en" messages={translationMessages.en}>
              <ProjectPageWithIntl
                {...props}
                projectCodes={fromJS([])}
                loadUsers={jest.fn()}
                loadingUsersError=""
                isAddPeopleFormOpen={false}
                closeAddPeopleForm={jest.fn()}
                openAddPeopleForm={jest.fn()}
                submitAddPeopleFormAndCloseIt={jest.fn()}
                usersNotInProject={fromJS([])}
                selectedProjectPeople={fromJS([])}
                loadProjectCodes={jest.fn()}
                updateField={jest.fn()}
                inlineProjectFormFields={fromJS({
                  name: { state: STATE_NORMAL, originalValue: 'test', value: 'test' },
                  description: { state: STATE_NORMAL, originalValue: 'test', value: 'test' },
                  startDate: { state: STATE_NORMAL, originalValue: null, value: null },
                  endDate: { state: STATE_NORMAL, originalValue: null, value: null },
                  code: { state: STATE_NORMAL, originalValue: null, value: null },
                  accountableId: { state: STATE_NORMAL, originalValue: null, value: null },
                })}
                updateProjectAttributesErrorMsg=""
                otherProjectClicked={jest.fn()}
                loadingProjectCodesError=""
                selectedTab={0}
                selectedTabChanged={jest.fn()}
                accountableOptions={[]}
                updateSelectedProjectId={jest.fn()}
                intl={{ formatMessage: jest.fn() }}
              />
            </IntlProvider>
          )
        }
        />
      </MemoryRouter>
    );
    expect(renderedComponent.html()).toMatchSnapshot();
  });
});
