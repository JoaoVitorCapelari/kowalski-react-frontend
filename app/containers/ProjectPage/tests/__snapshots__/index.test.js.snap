// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectPage /> renders 1`] = `"<div class=\\"Container__Container-dkRPdu cfjYIa\\"><div class=\\"Wrapper__Wrapper-cZxMuO hQAGbr\\"><div class=\\"ProjectSelectorWrapper__ProjectSelectorWrapper-csQcIx iYhmUQ\\"><div></div></div><div class=\\"ProjectPaneWrapper__ProjectPaneWrapper-ehuMBP erTuFr\\"><div></div><div class=\\"tabs ProjectPaneTabBar__TabsWrapper-cfNKjD duRwLX\\"><ul><li class=\\"is-active\\"><a role=\\"button\\" tabindex=\\"0\\">General</a></li><li class=\\"\\"><a role=\\"button\\" tabindex=\\"0\\">Activities</a></li></ul></div><div class=\\"TabContentWrapper__TabContentWrapper-kxUKT cOokgi\\"></div></div></div></div>"`;
